workspaces/liferay-customer-workspace/client-extensions/liferay-customer-etc-spring-boot/src/main/java/com/liferay/customer/JiraRestController.java

```diff
@@ -108,33 +145,151 @@ public ResponseEntity<String> get(@PathVariable("issueKey") String issueKey)
+						_jiraSecurityVulnerabilityFieldPartnerPublishingDate));
+				sb.append(" <= now()");
 			}
 			else {
-				sb.append(" AND 'Customer Publishing Date' <= now()");
+				sb.append(" AND ");
+				sb.append(
+					_getJQLCustomField(
+						_jiraSecurityVulnerabilityFieldCustomerPublishingDate));
+				sb.append(" <= now()");
+			}
+
+			if (Validator.isNotNull(filterAffectedVersions)) {
+				String[] versions = filterAffectedVersions.split(
+					StringPool.COMMA);
+
+				sb.append(" AND ");
+				sb.append(_FIELD_AFFECTED_VERSION);
+				sb.append(" in ('");
+				sb.append(StringUtil.merge(versions, "','"));
+				sb.append("')");
+			}
+
+			if (Validator.isNotNull(filterCategories)) {
+				String[] categories = filterCategories.split(StringPool.COMMA);
+
+				sb.append(" AND ");
+				sb.append(
+					_getJQLCustomField(
+						_jiraSecurityVulnerabilityFieldCategory));
+				sb.append(" in ('");
+				sb.append(StringUtil.merge(categories, "','"));
+				sb.append("')");
+			}
+
+			if (Validator.isNotNull(filterClassifications)) {
+				String[] classifications = filterClassifications.split(
+					StringPool.COMMA);
+
+				sb.append(" AND ");
+				sb.append(
+					_getJQLCustomField(
+						_jiraSecurityVulnerabilityFieldIssueClassification));
+				sb.append(" in ('");
+				sb.append(StringUtil.merge(classifications, "','"));
+				sb.append("')");
+			}
+
+			if (Validator.isNotNull(filterFixVersions)) {
+				String[] fixVersions = filterFixVersions.split(
+					StringPool.COMMA);
+
+				sb.append(" AND ");
+				sb.append(_FIELD_FIX_VERSION);
+				sb.append(" in ('");
+				sb.append(StringUtil.merge(fixVersions, "','"));
+				sb.append("')");
+			}
+
+			if (Validator.isNotNull(filterSeverities)) {
+				String[] severities = filterSeverities.split(StringPool.COMMA);
+
+				sb.append(" AND ");
+				sb.append(
+					_getJQLCustomField(
+						_jiraSecurityVulnerabilityFieldSeverity));
+				sb.append(" in ('");
+				sb.append(StringUtil.merge(severities, "','"));
+				sb.append("')");
 			}
 
 			if (Validator.isNotNull(keywords)) {
-				sb.append(" AND ('Customer Portal Summary' ~ ");
+				sb.append(" AND (");
+				sb.append(
+					_getJQLCustomField(
+						_jiraSecurityVulnerabilityFieldCustomerPortalSummary));
+				sb.append(" ~ ");
 				sb.append(StringUtil.quote(keywords));
-				sb.append(" OR 'CVE IDs' ~ ");
+				sb.append(" OR ");
+				sb.append(
+					_getJQLCustomField(_jiraSecurityVulnerabilityFieldCVEIds));
+				sb.append(" ~ ");
 				sb.append(StringUtil.quote(keywords));
 				sb.append(")");
 			}
 
+			sb.append(" ORDER BY ");
+
+			if (_isPartner()) {
+				sb.append(
+					_getJQLCustomField(
+						_jiraSecurityVulnerabilityFieldPartnerPublishingDate));
+			}
+			else {
+				sb.append(
+					_getJQLCustomField(
+						_jiraSecurityVulnerabilityFieldCustomerPublishingDate));
+			}
+
+			if (Validator.isNotNull(sortOrder)) {
+				sb.append(" ");
+				sb.append(sortOrder);
+			}
+			else {
+				sb.append(" DESC");
```

Comment:
can you make this a default value of the sortOrder requestparam?

Reply:
Re-pushed, can you check if I understood you right? I swapped the if statement. Thanks!

Reply:
sorry, i meant the defaultValue on the @RequestParam annotation

Reply:
my fault, I was doing all the default values and completely missed it, re-pushed the changes